### ISSUE TITLE
Tech Debt: local bug in mail/api.py send_batch

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/micromasters#3035]
I noticed this while reviewing nearby code. It is contingent on `MAILGUN_RECIPIENT_OVERRIDE` being set so this shouldn't affect production.

- Here we're consuming an iterable: `original_recipients_chunk`https://github.com/mitodl/micromasters/blob/617f80c7ccff8c8e6ad57a677c5c4fafec11b513/mail/api.py#L158
- In the case of `MAILGUN_RECIPIENT_OVERRIDE` being set, that value ends up being the original `recipients` generator passed into `send_batch` (see L140 and L114 in the same file).
- This causes the generator to be exhausted when we create the exception on L158 (and could, if the generator consumed enough memory, cause an OOM-kill) and then I would expect that `chunks` then stops iterating since we've mutated the same reference it is mutating.
- Additionally, the further downside is the logged exception wouldn't actually contain the set of emails from the chunk, but rather the remainder of emails from the generator that we hadn't sent or attempted to send yet.

TLDR: the exception logging should probably just unconditionally use `chunk`.
